cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -217,36 +159,95 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
-	// As an instructor, I can change the name of the assignment for my course.
-	@PutMapping("/gradebook/edit-name/{id}")
-	public void editAssignmentName (@RequestBody AssignmentDTO assignments, @PathVariable("id") Integer assignmentId ) {
-		
-		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
-		checkAssignment(assignmentId, email);  // check that user name matches instructor email of the course.
+	// As an instructor for a course , I can add a new assignment for my course.  
+	// The assignment has a name and a due date.
+	@PostMapping("/gradebook/add")
+	@Transactional
+	public AssignmentDTO addAssignment(@RequestBody AssignmentDTO assignmentDTO) {
+		Assignment duplicateAssignment = assignmentRepository.findById(assignmentDTO.assignmentId);
```

Comment:
I wouldn't expect that an assignment submitted to this endpoint would have an assignment id.
This may be unnecessary unless there's another unique field that needs checking for.